mods/ra/maps/exodus/exodus.lua

```diff
@@ -0,0 +1,398 @@
+	else
+		units = SovietUnits1
+	end
+
+	unit = { Utils.Random(units) }
+	Reinforcements.Reinforce(soviets, unit, route)
+
+	delay = math.max(attackAtFrame - 5, minAttackAtFrame)
+
+	Trigger.AfterDelay(delay, SpawnSovietUnits)
+end
+
+SovietParadrop = 0
+SendSovietParadrop = function()
+	local sovietParadrops = SovietParadrops[Difficulty]
+
+	if (SovietParadrop > sovietParadrops) then
+		return
+	end
+
+	SovietParadrop = SovietParadrop + 1
+
+	Utils.Do(humans, function(player)
+		Media.PlaySpeechNotification(player, "SovietForcesApproaching")
+	end)
+
+	local x = Utils.RandomInteger(ParadropBoxTopLeft.Location.X, ParadropBoxBottomRight.Location.X)
+	local y = Utils.RandomInteger(ParadropBoxBottomRight.Location.Y, ParadropBoxTopLeft.Location.Y)
+
+	local randomParadropCell = CPos.New(x, y)
+	local lz = Map.CenterOfCell(randomParadropCell)
+
+	local powerproxy = Actor.Create("powerproxy.paratroopers", false, { Owner = soviets })
+	powerproxy.SendParatroopers(lz)
+	powerproxy.Destroy()
+
+	Trigger.AfterDelay(sovietParadropTicks, SendSovietParadrop)
+end
+
+IdleHunt = function(unit)
+	Trigger.OnIdle(unit, unit.Hunt)
+end
+
+ManageSovietUnits = function()
+	Utils.Do(SovietRallyPoints, function(rallyPoint)
+		local radius = WDist.FromCells(8)
+		local units = Map.ActorsInCircle(rallyPoint.CenterPosition, radius, function(actor)
+			return actor.Owner == soviets and actor.HasProperty("Hunt")
+		end)
+
+		if #units > SovietGroupSize then
+			Utils.Do(units, IdleHunt)
+		end
+	end)
+
+	local scatteredUnits = Utils.Where(soviets.GetGroundAttackers(), function(unit)
+		return not Map.IsNamedActor(unit) and unit.IsIdle and unit.HasProperty("Hunt")
+	end)
+	Utils.Do(scatteredUnits, IdleHunt)
+end
+
+AircraftTargets = function()
+	local targets = Utils.Where(Map.ActorsInWorld, function(a)
+		return (a.Owner == allies1 or a.Owner == allies2)
+			and (a.HasProperty("Move") or a.HasProperty("StartBuildingRepairs"))
+	end)
+
+	return targets
+end
+
+YakAttack = function(yak)
+	local targets = AircraftTargets()
+	if #targets > 0 then
+		target = Utils.Random(targets)
+	else
+		yak.Wait(DateTime.Seconds(5))
+	end
+
+	if yak.AmmoCount() > 0 then
+		yak.Attack(target)
+	else
+		yak.ReturnToBase()
+		yak.Resupply()
+	end
+end
+
+ManageSovietAircraft = function()
+	if allies1.IsObjectiveCompleted(destroyAirbases) then
+		return
+	end
+
+	local maxSovietYaks = MaxSovietYaks[Difficulty]
+	local sovietYaks = soviets.GetActorsByType('yak')
+	if #sovietYaks < maxSovietYaks then
+		soviets.Build(Yak, function(units)
+			local yak = units[1]
+			Trigger.OnIdle(yak, YakAttack)
+		end)
+	end
+
+	if #sovietYaks > 0 then
+		local circling = Utils.Where(sovietYaks, function(yak) return yak.IsCircling end)
+		if #circling > 0 then
+			Utils.Do(circling, YakAttack)
+		end
+
+		local grounded = Utils.Where(sovietYaks, function(yak)
+			return yak.CenterPosition.Z < 1 and yak.AmmoCount() == yak.MaximumAmmoCount()
+		end)
+		if #grounded > 0 then
+			Utils.Do(grounded, YakAttack)
```

Comment:
This doesn't work because of [some weird code in `ResupplyAircraft`](https://github.com/OpenRA/OpenRA/blob/8233857be1e4684d0b7519760148404fecc0c8e8/OpenRA.Mods.Common/Activities/Air/ResupplyAircraft.cs#L39). My suggestion would be to simply insert a `yak.Stop()` before `yak.Attack(target)` in `YakAttack`.